Report lines starting with && in tidy

zbynekwinkler opened the pull request:
Partial implementation of the issue #10692 (the easy part).

<!-- Reviewable:start -->

---

This change is [<img src="https://reviewable.io/review_button.svg" height="35" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/servo/servo/10751)

<!-- Reviewable:end -->

zbynekwinkler committed:
Report lines starting with &&.

zbynekwinkler committed:
Add newline to the output.

The last line of output from tidy did not end with a newline if some
errors were reported.

highfive commented:
Thanks for the pull request, and welcome! The Servo team is excited to review your changes, and you should hear from @ms2ger (or someone else) soon.

highfive commented:
Heads up! This PR modifies the following files:
 * @wafflespeanut: python/tidy/servo_tidy_tests/test_tidy.py, python/tidy/servo_tidy_tests/rust_tidy.rs, python/tidy/servo_tidy/tidy.py

zwn commented:
I see the CI is blocking on tidy returning 0. What should be the next step now? Should I go and clean up the instances? I am reluctant to do that since it will mess up blame (by replacing relevant info with irrelevant indentation changes).

metajack commented:
If you change tidy you'll need to make tidy pass as well. You might make git blame slightly harder, but I think we all know how to follow history to get the real blame, so it's not a big concern.

Ms2ger commented:
And it's not even 20 lines, so *shrug*

zbynekwinkler committed:
Move '&&' to the end of the previous line.

Following https://github.com/servo/servo/issues/10692 this is just a
formating change to satisfy a new tidy requirement of not having '&&' at
the beginning of a line.

zwn commented:
Anything else?

bors-servo commented:
:umbrella: The latest upstream changes (presumably #10758) made this pull request unmergeable. Please resolve the merge conflicts.

Wafflespeanut commented on python/tidy/servo_tidy_tests/test_tidy.py:
```diff
@@ -61,6 +61,7 @@ def test_rust(self):
         self.assertEqual('extra space before :', errors.next()[2])
         self.assertEqual('use &[T] instead of &Vec<T>', errors.next()[2])
         self.assertEqual('use &str instead of &String', errors.next()[2])
+        self.assertEqual('operators should go at the end of the first line', errors.next()[2])
```
Now that #10758 has landed, we should ensure that there are no more errors.

zwn replied:
Done.

zwn commented:
I've rebased it on current master and reordered the commits so that the check is enabled only after the files had been fixed.

Wafflespeanut commented:
Thanks for doing this! :)

@bors-servo r+

bors-servo commented:
:pushpin: Commit 39780ca has been approved by `Wafflespeanut`

<!-- @bors-servo r=Wafflespeanut 39780ca39f2beff6c0f8e55eab75022b27cd046e -->

bors-servo commented:
:hourglass: Testing commit 39780ca with merge d926b5d...

bors-servo commented:
:sunny: Test successful - [android](http://build.servo.org/builders/android/builds/466), [arm32](http://build.servo.org/builders/arm32/builds/498), [arm64](http://build.servo.org/builders/arm64/builds/483), [linux-dev](http://build.servo.org/builders/linux-dev/builds/493), [linux-rel](http://build.servo.org/builders/linux-rel/builds/543), [mac-dev-unit](http://build.servo.org/builders/mac-dev-unit/builds/507), [mac-rel-css](http://build.servo.org/builders/mac-rel-css/builds/493), [mac-rel-wpt](http://build.servo.org/builders/mac-rel-wpt/builds/520)
